docs/fsharp/language-reference/caller-information.md

```diff
@@ -0,0 +1,84 @@
+
+To obtain this information, you use attributes that are applied to optional parameters, each of which has a default value. The following table lists the Caller Info attributes that are defined in the [System.Runtime.CompilerServices](https://docs.microsoft.com/dotnet/api/system.runtime.compilerservices) namespace:
+
+|Attribute|Description|Type|
+|---------|-----------|----|
+|[CallerFilePath](https://docs.microsoft.com/dotnet/api/system.runtime.compilerservices.callerfilepathattribute)|Full path of the source file that contains the caller. This is the file path at compile time.|`String`
+|[CallerLineNumber](https://docs.microsoft.com/dotnet/api/system.runtime.compilerservices.callerlinenumberattribute)|Line number in the source file at which the method is called.|`Integer`|
+|[CallerMemberName](https://docs.microsoft.com/dotnet/api/system.runtime.compilerservices.callermembernameattribute)|Method or property name of the caller. See the Member Names section later in this topic.|`String`|
+
+## Example
+
+The following example shows how you might use these attributes to trace a caller.
+
+```fsharp
+open System.Diagnostics
+open System.Runtime.CompilerServices
+
+type Tracer() =
+    member __.DoTrace(msg: string,
+                      [<CallerMemberName>] ?memberName: string,
+                      [<CallerFilePath>] ?path: string
+                      [<CallerLineNumber>] ?line: int) =
+        Trace.WriteLine(sprintf "Message: %s" message)
+        match (memberName, path, line) with
+        | Some m, Some p, Some l ->
+            Trace.WriteLine(sprintf "Member name: %s" m)
+            Trace.WriteLine(sprintf "Source file path: %s" p)
+            Trace.WriteLine(sprintf "Source line number: %d" l)
+        | _,_,_ -> ()
+```
+
+## Remarks
+
+You must specify an explicit default value for each optional parameter. You can't apply Caller Info attributes to parameters that aren't specified as optional.
+
+The Caller Info attributes don't make a parameter optional. Instead, they affect the default value that's passed in when the argument is omitted.
+
+Caller Info values are emitted as literals into the Intermediate Language (IL) at compile time. Unlike the results of the [StackTrace](https://docs.microsoft.com/dotnet/api/system.diagnostics.stacktrace) property for exceptions, the results aren't affected by obfuscation.
+
+You can explicitly supply the optional arguments to control the caller information or to hide caller information.
+
+## Member Names
+
+You can use the `CallerMemberName` attribute to avoid specifying the member name as a `String` argument to the called method. By using this technique, you avoid the problem that Rename Refactoring doesn't change the `String` values. This benefit is especially useful for the following tasks:
```

Comment:
link to the APIs? if you want to use a different name than the default class name, you can link like this:
[CallerMemberName](xref:System.Runtime.CompilerServices.CallerMemberNameAttribute)